install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -1076,6 +1076,19 @@ hubble:
       #   --set hubble.redact.enabled="true"
       #   --set hubble.redact.http.urlQuery="true"
       urlQuery: false
+      # -- Enables redacting user info, e.g., password when basic auth is used.
+      # Example:
+      #
+      #   redact:
+      #     enabled: true
+      #     http:
+      #       userInfo: true
+      #
+      # You can specify the options from the helm CLI:
+      #
+      #   --set hubble.redact.enabled="true"
+      #   --set hubble.redact.http.userInfo="true"
+      userInfo: true
```

Comment:
Usually I prefer to keep new behaviour disabled by default, but this has security implications.

@lambdanis what are your thoughts?

Reply:
Hey! About the default values it was already discussed in the original issue with @rolinh and others. See item 2 at https://github.com/cilium/cilium/issues/23887#issuecomment-1697655676. 
In general we will need to revisit and decide what are the best defaults for the whole "`hubble-redact`" feature. This can happen here or in a follow up PR.

Reply:
Considering that it's not a new functionality, but this PR just allows us to disable it, enabling it be default seems the right approach.

Reply:
Yup, it's the old behaviour, just now it's configurable.
@ChrsMark Changing the defaults is a breaking change, so I think it should be a separate PR (feel free to open an issue too if there isn't one already), to clearly document the motivations.

Reply:
As @tommyp1ckles mentions, since this has security implications, IMHO we should preserve existing behavior, that is, redact by default.

Still, let's discuss this as a whole and come to a decision about sane default behavior for all redact options provided by Hubble.

Reply:
this was resolved but I still see that we're enabling by default, is the plan to move this to a separate PR as @lambdanis mentioned?

Reply:
Based on @gandro 's and @lambdanis 's comments above, my understanding is:

1. Preserve existing behavior in this PR, that is, enable the newly added option by default. If we don't, we break security until the follow-up PR
2. Decide on sane default values for all Hubble redact options in a follow-up PR (it might take a while to arrive at decisions for all options)

LMK if you want me to disable the new option in this PR, I can always push a FIXUP.